Nikita-Stankevichyus/task_2_hashToArray.js

```diff
@@ -0,0 +1,30 @@
+"use strict";
+
+var hashToArray = function convertHashToArray(hash) {
+
+  // function is immutable
+  var workHash = hash;
+  
+  var ownProps = Object.getOwnPropertyNames(workHash);
+  
+  // we know for sure how many elements the result array will contain
+  var resultArray = new Array(ownProps.length);
+
+
+  for(var i = 0; i < ownProps.length; i++) {
+    
+    // first evaluate an element of the array as array itself
+    resultArray[i] = [];
+
+    // the first is the key
+    resultArray[i][0] = ownProps[i];
+
+    // the second is the value
+    resultArray[i][1] = workHash[ownProps[i]]; 
+
+  }
+
+
+  return resultArray;
+  
+}
```

Comment:
> 
> 
> @Stanoque Works as expected
> Please try to write less code - it's will help to manage it in future.
> So, here we can use, for ex., `resultArray.push([ownProps[i], workHash[ownProps[i]]])` instruction instead of current variant

Code corrected!